### DISCUSSION TITLE
HEXO全新安装ICARUS报错

### DISCUSSION THREAD

**Discussion by qvsijia** [ppoffice/hexo-theme-icarus#1091] [General]
本机环境：
nodejs v18.4.0
npm v8.14.0
git v2.37.1

hexo-cli: 4.3.0
os: win32 10.0.22000
node: 18.4.0
v8: 10.2.154.4-node.8
uv: 1.43.0
zlib: 1.2.11
brotli: 1.0.9
ares: 1.18.1
modules: 108
nghttp2: 1.47.0
napi: 8
llhttp: 6.0.6
openssl: 3.0.3+quic
cldr: 41.0
icu: 71.1
tz: 2022a
unicode: 14.0
ngtcp2: 0.1.0-DEV
nghttp3: 0.1.0-DEV

在新的干净环境下，使用npm安装之后，打开localhost:4000就是下面报错那样

报错如下：
const { Component } = require('inferno'); const classname = require('hexo-component-inferno/lib/util/classname'); const Head = require('./common/head'); const Navbar = require('./common/navbar'); const Widgets = require('./common/widgets'); const Footer = require('./common/footer'); const Scripts = require('./common/scripts'); const Search = require('./common/search'); module.exports = class extends Component { render() { const { site, config, page, helper, body } = this.props; const language = page.lang || page.language || config.language; const columnCount = Widgets.getColumnCount(config.widgets, config, page); return
; } };

**Comment by ppoffice** [comment]
请尝试`npm install hexo-component-inferno@1.2.0`。如果不行的话，请尝试从源码安装Icarus：https://ppoffice.github.io/hexo-theme-icarus/uncategorized/icarus%E5%BF%AB%E9%80%9F%E4%B8%8A%E6%89%8B/

**Reply by qvsijia** [reply]
1、仅 **npm install hexo-component-inferno@1.2.0** 无效
2、仅 **git clone https://github.com/ppoffice/hexo-theme-icarus.git themes/icarus -b <version number> --depth 1** 无效
3、步骤1+2，问题解决
谢谢大佬回复

**Reply by qvsijia** [reply]
之后我又测试了一下，仅git clone是不行的，缺少了好几个npm项目。
然后我在一个新项目里：
1、npm install hexo-theme-icarus
2、git clone https://github.com/ppoffice/hexo-theme-icarus.git themes/icarus --depth 1
3、npm install hexo-component-inferno@1.2.0
然后问题就解决了

**Reply by ppoffice** [reply]
等icarus新版本发布，就不需要从git安装了。
